src/Modules/Profile/Commands.Profile/InvokePowerBIRestMethod.cs

```diff
@@ -215,6 +219,15 @@ protected virtual void PopulateClient(IAccessToken token, HttpClient client)
             client.DefaultRequestHeaders.UserAgent.Clear();
             client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("MicrosoftPowerBIMgmt-InvokeRest", PowerBICmdlet.CmdletVersion));
 
+            if (this.TimeoutSec < 0)
+            {
+                this.Logger.ThrowTerminatingError($"{nameof(this.TimeoutSec)} value cannot be negative.");
+            }
+            else if (this.TimeoutSec > 0)
+            {
+                client.Timeout = TimeSpan.FromSeconds(this.TimeoutSec);
+            }
+
```

Comment:
Try to differentiate no input (use default 100s) vs input=0(use infinite timeout as powershell document)

Reply:
Updated with latest commit